Makefile

```diff
@@ -33,7 +35,7 @@ help:
 setup: clean virtualenv requirements test-requirements
 
 run:
-	python3 $(CURDIR)/eks_rolling_update.py --help
+	source $(VENV)/bin/activate && $(PYTHON_PATH) $(CURDIR)/eks_rolling_update.py $(ARGS)
```

Comment:
Check out how we do the venv thing in all data projects, e.g. https://github.com/mx51/data-glue-extractor/blob/master/Makefile#L5-L11

Reply:
Thanks @gouline. I jumped into the glue repo and ran `make` which worked fine. But then when I ran `make requirements` it installed the packages in my home default path `/home/nikola/.local/lib/python3.10/site-packages`, not in the venv. But If I run `source .venv/bin/activate` before `make requirements`, it will load the venv and install the modules within the venv `.venv/lib/python3.10/site-packages`.

Reply:
You have to start every `make` invocation with `venv` to use the virtual environment, e.g. `make venv requirements`. If somebody running this project already has dependencies installed locally, they don't want to create another virtual environment just to run your project. Hence they can run `make venv requirements run` or `make requirements run`.

Reply:
If you look at what the `venv` target does, it still sources the activate script, it just avoids having to do that in every command invocation throughout the script, as you have here. Alternatively, if you want to enforce running `venv` for everyone, then you can just make it a dependency for your targets, e.g. `run: venv`.

Reply:
Ah yep. Got my head around it now. I have made the updates to the make file, however, since this is a fork from the public project https://github.com/hellofresh/eks-rolling-update it will be up to the code owners to approve the changes. I am still happy to commit the updates in our fork.